### PR TITLE
up now fully leverages libcompose (closes #82, closes #89)

### LINKED ISSUE #82: Support use of variable substitution in docker-compose.yml for things other than envvars
[Variable substitution](https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution) is currently only supported for [envvars](https://docs.docker.com/compose/environment-variables/).

```yaml
environment:
  RACK_ENV: development
  SHOW: "true"
  DB_PASSWORD: ${DB_PASSWORD}
```

This would add support for other things (currently [`image`](https://github.com/turnerlabs/harbor-compose/blob/master/compose-reference.md#image) and [`ports`](https://github.com/turnerlabs/harbor-compose/blob/master/compose-reference.md#ports)).

```yaml
web:
  image: "webapp:${TAG}"
```

### LINKED ISSUE #89: HEALTHCHECK environment variable not found when in an included env_file
When HEALTHCHECK is in a file specified in env_file: in docker-compose.yml it is not found by harbor-compose, which reports the error that HEALTHCHECK must be specified.

### PR DESCRIPTION
**Pull request by jritsema**
The `up` command now supports things like...

.env
```
VERSION=0.0.13-rc.42
PORT=3000
HEALTHCHECK=/health
```

docker-compose.yml
```
version: "2"
services:
  web:
    image: registry.services.dmtio.net/mss-poc-thingproxy:${VERSION}
    ports:
    - 80:${PORT}
    environment:
      PORT: ${PORT}
      HEALTHCHECK: ${HEALTHCHECK}
```

### CODE REVIEW

File: cmd/up_test.go [hunk 1]
```diff
@@ -37,3 +43,361 @@ services:
...
+    - app
+    replicas: 2
+    group: mss
+    property: turner
+    project: project
+    product: product
+`
+
+	//write .env file containing env vars, to current directory
+	err := ioutil.WriteFile(".env", []byte("FROM_DOT_ENV=foo"), 0644)
```
Comment (jkurz) [inline cmd/up_test.go:226]: very cool way to test this out.

**Review by jkurz — COMMENTED** [review]
One test is failing for me locally. Do we want to update this before we merge?

```
--- FAIL: TestDeleteFileSucceed (0.00s)
        Error Trace:    logout_test.go:11
	Error:      	Not equal:
	            	expected: false
	            	received: true
        Error Trace:    logout_test.go:12
	Error:      	Expected nil, but got: &os.PathError{Op:"remove", Path:"~/.harbor", Err:0x42}
FAIL
```

File: cmd/up_test.go (hunk 1 above)
Comment (jritsema) [inline cmd/up_test.go]: login/logout tests have been fixed

**Comment by jritsema** [comment]
login/logout tests have been fixed.  Can you re-test?

**Review by jkurz — APPROVED** [review]

**Comment by jkurz** [comment]
LGTM. Nice work.

**Pull request merged**